rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java

```diff
@@ -2761,7 +2768,7 @@ public Object execIdCall(
                 return realThis(thisObj, f).toString();
 
             case Id_exec:
-                return realThis(thisObj, f).execSub(cx, scope, args, MATCH);
```

Comment:
I see that we're also changing stuff in IdScriptableObject that's been there a while, so I'm trying to follow this. What is changing here that's requiring us to treat the name of the function object differently?

Reply:
The reason is that, in `IdScriptableObject`, the method `ensureType` (used by `realThis`) took an `IdFunctionObject`. However, that parameter was only used to get the function name. So, I've refactored it and added an overload that only takes the function name.
This allows me to have a method `NativeRegExp::js_exec` that does not go through the `IdFunctionObject` at all, and so it can easily be called from `NativeRegExpStringIterator` to implement point 3 of the spec algorithm [RegExpExec](https://tc39.es/ecma262/multipage/text-processing.html#sec-regexpexec).